x-pack/plugins/asset_manager/server/routes/utils.ts

```diff
@@ -16,3 +21,26 @@ export async function getClientsFromContext<T extends RequestHandlerContext>(con
     savedObjectsClient: coreContext.savedObjects.client,
   };
 }
+
+type ValidateStringAssetFiltersReturn =
+  | [{ error: RouteValidationError }]
+  | [null, AssetFilters | undefined];
+
+export function validateStringAssetFilters(
+  q: any,
+  res: RouteValidationResultFactory
+): ValidateStringAssetFiltersReturn {
+  if (typeof q.stringFilters === 'string') {
+    try {
+      const parsedFilters = JSON.parse(q.stringFilters);
+      if (assetFiltersSingleKindRT.is(parsedFilters)) {
```

Comment:
I thought this would return false if any properties of the parsedFilters don't exist on the type but it still returns true. For example `GET /api/asset-manager/assets/hosts?from=now-1m&to=now&stringFilters=%7B%22foobar%22%3A%22baz%22%7D` (`filters={"foobar": "baz"}`) passes through and returns all the services. Is this condition checking something else ?

Reply:
nvm, remembered we discussed this offline and this will validate that the overlapping properties are of the correct type but it won't throw on non-defined properties

Reply:
Yeah I removed the checking of unknown fields because it felt overly complex for what we need right now. We can add it back if we decide we need it or, maybe, if we move to that other runtime types lib that has it natively.